contrib/coredns/policy/dns.go

```diff
@@ -115,44 +132,45 @@ func resetTTL(r *dns.Msg) *dns.Msg {
 
 func (p *policyPlugin) setRedirectQueryAnswer(ctx context.Context, w dns.ResponseWriter, r *dns.Msg, dst string) (int, error) {
 
-	qName, qClass := getNameAndClass(r)
-
-	ip := net.ParseIP(dst)
-	rr := ip2rr(ip, qName, qClass)
-	if rr == nil {
-
-		dst = dns.Fqdn(dst)
-		rr = &dns.CNAME{
-			Hdr: dns.RR_Header{
-				Name:   qName,
-				Rrtype: dns.TypeCNAME,
-				Class:  qClass,
-			},
-			Target: dst,
+	var rr dns.RR
+	qName, qType, qClass := getNameTypeAndClass(r)
+	record := findRecord(dst, qType)
+	switch qType {
+	case dns.TypeA, dns.TypeAAAA, dns.TypeNone:
```

Comment:
`typeNone` is returned when `r==nil || len(r.Question)<=0` . In such case, we still send A/AAAA if ip4/ip6 in redirect string but otherwise resolve by CNAME and error out , like exisitng as well.

Reply:
Ok, i just wanted to check this because in existing function, we do not send error if querytype = A/AAAA and r == nil. Anyways will update now.